src/main/java/org/folio/inventory/resources/Items.java

```diff
@@ -107,6 +115,57 @@ public void register(Router router) {
           routingContext, webContext));
   }
 
+  private void move(RoutingContext routingContext) {
+
+    WebContext context = new WebContext(routingContext);
+    JsonObject itemsMoveJsonRequest = routingContext.getBodyAsJson();
+
+    Optional<ValidationError> validationError = itemsMoveHasRequiredFields(itemsMoveJsonRequest);
+    if (validationError.isPresent()) {
+      unprocessableEntity(routingContext.response(), validationError.get());
+    } else {
+      ItemsMove itemsMove = MoveUtil.jsonToItemsMove(itemsMoveJsonRequest);
+      List<String> itemIdsToUpdate = itemsMove.getItemIds();
+      try {
+        storage.getItemCollection(context).findByCql(
+          buildQueryByIds(itemIdsToUpdate),
+          PagingParameters.defaults(),
+          success -> updateManyItemsWithNewHoldingsRecordId(routingContext, context, success.getResult(), itemsMove.getToHoldingsRecordId(), itemIdsToUpdate),
+          FailureResponseConsumer.serverError(routingContext.response()));
+      } catch (UnsupportedEncodingException e) {
+        ServerErrorResponse.internalError(routingContext.response(), e.toString());
+      }
+    }
+  }
+
+  private void updateManyItemsWithNewHoldingsRecordId(
+    RoutingContext routingContext,
+    WebContext context,
+    MultipleRecords<Item> itemsToUpdate,
+    String toHoldingsRecordId,
+    List<String> itemIdsToUpdate) {
+
+    ItemCollection storageItemCollection = storage.getItemCollection(context);
+
+    List<CompletableFuture<Item>> updates = itemsToUpdate.records.stream()
```

Comment:
There is new batch API for item update `/item-storage/batch/synchronous` (use `?upsert=true`)

Reply:
@bohdan-suprun @marcjohnson-kint I have concern on it. Looks like this API works like transaction: it updates all or nothing from items list [item-sync.html](https://s3.amazonaws.com/foliodocs/api/mod-inventory-storage/p/item-sync.html). But we need to know what items aren't updated. What do you think on this?

Reply:
I believe that is true, it was decided that those batch APIs would be all or nothing. That may well make them inappropriate for use in this case, and mean the records have to be updated individually.